src/core/namespace.c

```diff
@@ -873,6 +873,15 @@ static int mount_procfs(const MountEntry *m, const NamespaceInfo *ns_info) {
          * mounted on /proc/ first. */
```

Comment:
The comment above needs to be updated and moved, as it states that /proc is always remounted. For example I'd add something like this:

"In case of unprivileged user namespaces (systemd --user), mounting /proc is not allowed by the kernel."

(the manpages of user_namespaces state that it is allowed with just a mount namespace and a username namespace, but I do not believe that is true, a PID namespace is needed as well at a minimum)